src/ApiCompat/Microsoft.DotNet.ApiCompatibility/AssemblySymbolLoader.cs

```diff
@@ -253,30 +254,31 @@ private IReadOnlyList<MetadataReference> LoadFromPaths(IEnumerable<string> paths
             List<MetadataReference> result = new();
             foreach (string path in paths)
             {
-                string? directory = null;
-
                 if (Directory.Exists(path))
                 {
-                    directory = path;
+                    // If a directory is passed in as a path, add that to the reference paths.
+                    // Otherwise, if a file is passed in, add its parent directory to the reference paths.
+                    _referencePathDirectories.Add(path);
+
                     foreach (string assembly in Directory.EnumerateFiles(path, "*.dll"))
                     {
                         result.Add(CreateOrGetMetadataReferenceFromPath(assembly, referenceAssemblyNamesToIgnore));
                     }
                 }
                 else if (File.Exists(path))
                 {
-                    directory = Path.GetDirectoryName(path);
+                    string? directory = Path.GetDirectoryName(path);
+                    // If a directory is passed in as a path, add that to the reference paths.
+                    // Otherwise, if a file is passed in, add its parent directory to the reference paths.
+                    if (!string.IsNullOrEmpty(directory))
```

Comment:
is there actually any scenario where a file exists but we can't determine its directory or is this more a defense-in-depth check?

Reply:
Path.GetDirectoryName returns a nullable string which made me check for it for two reasons:
1. Make the compiler happy (lol)
2. Even though the code doesn't look like it would ever return null with these given inputs, this could theoretically change as the method already returns a nullable string.

So yes, a defense-in-depth check.